nnvm/src/top/tensor/transform.cc

```diff
@@ -830,6 +830,100 @@ Examples::
     };
 });
 
+// strided_slice
+DMLC_REGISTER_PARAMETER(StridedSliceParam);
+
+inline void StridedSliceParamParser(nnvm::NodeAttrs* attrs) {
+  StridedSliceParam param;
+  param.Init(attrs->dict);
+  attrs->parsed = std::move(param);
+}
+
+inline bool StridedSliceInferShape(const NodeAttrs& attrs,
+                            std::vector<TShape>* in_shape,
+                            std::vector<TShape>* out_shape) {
+  const StridedSliceParam& param = nnvm::get<StridedSliceParam>(attrs.parsed);
+  const TShape& dshape = (*in_shape)[0];
+  if (dshape.ndim() == 0) return false;
+  TShape oshape = dshape;
+  dim_t num_axis = dshape.ndim();
+
+  std::vector<int64_t> begin_vec;
+  std::copy(param.begin.begin(), param.begin.end(), std::back_inserter(begin_vec));
+  for (dim_t i = begin_vec.size(); i < num_axis; ++i) {
+    begin_vec.push_back(0);
+  }
+
+  std::vector<int64_t> end_vec;
```

Comment:
Handle INT_MAX for end

Reply:
I had considered this scenario while implementation, and concluded that the functionality can be achieved using the current implementation, which i did in case of Tensorflow case(had many other parameters like ellipsis_mask, new_axis_mask, shrink_axis_mask)(PR coming soon), and it works fine, similarly can be done for ONNX also.

NOTE: in Tensorflow ellipsis_mask is nothing but similar to axis input case in ONNX.

I have gone through onnx slice operator, my understanding is INT_MAX they will use when the size in the particular dimension is unknown, which is not the case in NNVM.

Apart from that our current implementation can handle INT_MAX input perfectly, it will strip down end value to the max size of the dimension. In my opinion, we should not modify the base topi operators based on each frontends, provided the functionality can be achieved by modifying the inputs to the operator.

Please let me know @tqchen , @srkreddy1238 , @FrozenGene , your opinions. Will wait for your feedback.

Reply:
I don't think apply INT_MAX for slice op making sense. INT_MAX is for unknown dimension, which is not the things NNVM should consider.  I translated many models with unknown dimension (such as ssd-mobilenet / Deeplab V3 tensorflow model's preprocess input), for this situation, the users should provide the concrete shape, not leaving this to NNVM.

Reply:
It's not the dimension which is unknown here. It's the range inside the dimension.

INT_MAX here indicates to use max available range from input.